lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +178,141 @@ const steps = {
       )}\n`
     );
   },
+  // TODO: TESTS
+  handleProviderCreation: async ({ configuration: { org: orgName } }) => {
+    const providersUrl = `https://app.serverless.com/${orgName}/settings/providers?source=cli`;
+    openBrowser(providersUrl);
+    // TODO: CHANGE MESSAGING
+    process.stdout.write('Waiting for creation of provider...\n');
+
+    let onEvent;
+
+    const p = new Promise((resolve) => {
+      let inquirerPrompt;
+
+      const timeoutDuration = 1000 * 60; // 1 minute
+      const showSkipPromptTimeout = setTimeout(() => {
+        inquirerPrompt = inquirer.prompt({
+          message: '\n Press Enter to continue without setting up provider',
+          name: 'skipProviderSetup',
+        });
+
+        inquirerPrompt.then(() => resolve(null));
+      }, timeoutDuration);
+
+      onEvent = (event) => {
+        if (inquirerPrompt) {
+          // Disable inquirer prompt asking to skip without setting provider
+          inquirerPrompt.ui.close();
+        }
+
+        clearTimeout(showSkipPromptTimeout);
+        resolve(event);
+      };
+    });
+
+    // Listen for `provider.created` event to detect creation of new provider
+    const sdk = await getSdkInstance(orgName);
+    await sdk.connect({
+      orgName,
+      onEvent,
+      filter: {
+        events: ['provider.created'],
+      },
+    });
+
+    let maybeEvent;
+    try {
+      maybeEvent = await p;
+    } finally {
+      sdk.disconnect();
+    }
+
+    if (maybeEvent) {
+      process.stdout.write(`\nDetected creation of provider: "${maybeEvent.data.object.alias}"\n`);
+    } else {
+      // TODO: ADD MESSAGING THAT YOU SKIPPED SETUP OF THE PROVIDER
+    }
+
+    // TODO: CONSIDER IF WE SHOULD CREATE DIRECT LINK for default provider
+  },
+  // TODO: TESTS
+  linkProviderToService: async (app, service, orgName, providerUid) => {
+    const sdk = await getSdkInstance(orgName);
+    const linkType = 'service';
+    const linkUid = getProviderLinkUid(app, service);
+    const orgUid = await getOrgUidByName(orgName);
+    return sdk.createProviderLink(orgUid, linkType, linkUid, providerUid);
+  },
 };
 
 module.exports = {
-  async isApplicable({ configuration }) {
+  async isApplicable(context) {
+    const { configuration } = context;
     if (
       _.get(configuration, 'provider') !== 'aws' &&
       _.get(configuration, 'provider.name') !== 'aws'
     ) {
       return false;
     }
     if (new AWS.S3().config.credentials) return false;
-    return !(await awsCredentials.resolveFileProfiles()).size;
+    if ((await awsCredentials.resolveFileProfiles()).size) return false;
+
+    // TODO: TESTS
+    const orgName = _.get(configuration, 'org');
+    if (orgName) {
```

Comment:
That's a great question - I forgot about `SERVERLESS_ACCESS_KEY` - I will see what we can do about it

Reply:
As mentioned it will be possible to handle `SERVERLESS_ACCESS_KEY` - the changes in `set-dashboard-org` will be implement to correctly recognize it

Reply:
Yes, adapting to support for `SERVERLESS_ACCESS_KEY` won't require any changes in this step - I will follow up with a PR in `dashboard-plugin`.

Reply:
hmm.. but `dashboard-plugin` is not invokved anywhere here. It's invoked only in context of `Serverless` instance, while here we're not working with `Serveless` instance.

Technically here to resolve `org` reliably, we should probably imply a logic as:

```javascript
if (SERVERLESS_ACCESS_KEY) org = resolveFrom(SERVERLESS_ACCESS_KEY);
else org = options.org || configuration.org
```

(?)

Reply:
It depends on the implementation of `dashboard-set-org` step - it doesn't have to be invoked explicitly, as the `org` will be populated in configuration. I don't think we should even consider other situations, because it `org` was not added to config it won't be able to use `provider` credentials. Am I missing something here?

Reply:
`dashboard-set-org` should update `configuration.org` only if it was updated in physical file, and I don't think that just using `SERVERLESS_ACCESS_KEY` env variable should trigger configuration rewrite mechanism.

Also best if we minimize dependency among steps (their order etc). Ideally they all should work exclusively (when possible)

Reply:
> Also best if we minimize dependency among steps
I wouldn't call it a dependency. From the perspective of this step, in my opinion, we should not even care if `SERVERLESS_ACCESS_KEY` is used, because it's a mechanism to authenticate to Dashboard (so resolved during `login/register` step). Similarily for `configuration.org` - if it's not present, I don't think it makes sense to show this step at all (or only offer the option to configure local credentials). 

> dashboard-set-org should update configuration.org only if it was updated in physical file

I think I'm not following here. Could you clarify what do you mean? In `dashboard-set-org` it's either set automatically if user logged in and only have one organization or user selects it from the list and then the configuration is updated. Setting `SERVERLESS_ACCESS_KEY` is in my opinion an explicit way to essentially "override" both `login/register` and `org` selection part from `dashboard-set-org`. What do you think the behavior should be then?

Reply:
We reiterated offline and agreed that: `configuration.org` has to be set in serverless.yml for dashboard functionality to jump in, and that it should reflect `SERVERLESS_ACCESS_KEY` (if not, dashboard will fail with meaningful error), therefore we can skip here `SERVERLESS_ACCESS_KEY` handling.

However when confirming on providers setup applicability, we should additionally confirm that user is logged in _or_ `SERVERLESS_ACCESS_KEY` is provided. If it's not the case, we should not provider providers option (currently there's just `configuration.org` check)